### PR TITLE
Cmssw 8 0 x new b2 g ana fw dev

### PR DESCRIPTION
**Pull request by tmitchel**

### CODE REVIEW

File: plugins/OS2LAna.cc [hunk 1]
```diff
@@ -720,19 +722,27 @@ bool OS2LAna::filter(edm::Event& evt, const edm::EventSetup& iSetup) {
   h1_["sqrtChi2"] ->Fill(sqrt(chi2_result.first), evtwt);
   h1_["chi2_mass"] ->Fill(chi2_result.second, evtwt);
 */
+  std::auto_ptr<double> ptr_fevtwt ( new double(evtwt) ) ;
+  std::auto_ptr<double> ptr_evtwt ( new double(presel_wt) ) ; 
+  evt.put(ptr_fevtwt, "finalWeight");
+  evt.put(ptr_evtwt, "PreWeight");
+
```
Comment (dmajumder) [inline plugins/OS2LAna.cc:729]: @tmitchel should't the above lines be inside the if (skim_)?

Comment (tmitchel) [reply plugins/OS2LAna.cc:729]: I don't need it for the skim. I want it for when I'm running on skims to store the weight for a later module.

Just like eventcleaner stores the weight to be read in os2lana

Comment (dmajumder) [reply plugins/OS2LAna.cc:729]: @tmitchel this is the massReco module?

Comment (tmitchel) [reply plugins/OS2LAna.cc:729]: yes

File: test/os2lana_cfg.py [hunk 2]
```diff
@@ -291,6 +299,7 @@
     *process.evtcleaner
     *process.cleanedEvents
     *process.ana
+#    *process.massReco
     *process.finalEvents
     )
```
Comment (dmajumder) [inline test/os2lana_cfg.py:304]: @tmitchel when you run the mass reco, you always run on the skims, right?

in that case, do you need to run all the other modules in your path every time?

Comment (tmitchel) [reply test/os2lana_cfg.py:304]: I do because I want to read the weights that are calculated in evtcleaner and modified in ana

**Pull request closed without merging**